crates/iota-data-ingestion/src/workers/archival.rs

```diff
@@ -186,8 +186,8 @@ impl Reducer<ArchivalWorker> for ArchivalReducer {
 
     fn should_close_batch(
         &self,
-        batch: &[CheckpointData],
-        next_item: Option<&CheckpointData>,
+        batch: &[Arc<CheckpointData>],
+        next_item: Option<&Arc<CheckpointData>>,
```

Comment:
This seems odd. I would expect
```rust
        next_item: Option<Arc<CheckpointData>>,
```
Shouldn't you refactor the `Reducer` trait as well to accept `Arc` instead of `&`?

Reply:
As discussed yesterday, I've tried to find some other solution but without any tangible success. I would like to propose to leave it for now and resume it in future iterations.